include/alpaka/warp/WarpUniformCudaHipBuiltIn.hpp

```diff
@@ -113,34 +113,75 @@ namespace alpaka::warp
         template<>
         struct Shfl<WarpUniformCudaHipBuiltIn>
         {
-            //-------------------------------------------------------------
+            template<typename T>
             __device__ static auto shfl(
                 [[maybe_unused]] warp::WarpUniformCudaHipBuiltIn const& warp,
-                float val,
+                T val,
                 int srcLane,
-                std::int32_t width) -> float
+                std::int32_t width) -> T
             {
 #        if defined(ALPAKA_ACC_GPU_CUDA_ENABLED)
                 return __shfl_sync(activemask(warp), val, srcLane, width);
```

Comment:
I know that this was already there, but shouldn't this be
```c++
return __shfl_sync(0xffffffff, val, srcLane, width);
```
instead ?

Possible preceded by a `__syncwarp()` ?

Reply:
This comes from https://github.com/alpaka-group/alpaka/pull/1273.
@psychocoderHPC what do you think ?

Reply:
For example, HIP does
```c++
#if CUDA_VERSION >= CUDA_9000
#define __shfl(...)      __shfl_sync(0xffffffff, __VA_ARGS__)
#define __shfl_up(...)   __shfl_up_sync(0xffffffff, __VA_ARGS__)
#define __shfl_down(...) __shfl_down_sync(0xffffffff, __VA_ARGS__)
#define __shfl_xor(...)  __shfl_xor_sync(0xffffffff, __VA_ARGS__)
#endif // CUDA_VERSION >= CUDA_9000
```

Reply:
To be consistent with other backends may be `return __shfl_sync(0xffffffff, val, srcLane, width);` but I am not fully sure.

Reply:
I'll make a PR for this.